core/chains/evm/headtracker/head_listener.go

```diff
@@ -28,32 +32,62 @@ var (
 	}, []string{"evmChainID"})
 )
 
-type headListener struct {
-	config           Config
-	ethClient        evmclient.Client
+type headListener[
+	H commontypes.Head[BLOCK_HASH],
+	HTH htrktypes.Head[H, BLOCK_HASH, ID],
+	S commontypes.Subscription,
+	ID txmgrtypes.ID,
+	BLOCK_HASH commontypes.Hashable,
+	CLIENT htrktypes.Client[HTH, S, ID, BLOCK_HASH],
```

Comment:
This seems wrong to use HTH here.
The definition of Cleint interfaces I see goes like this:

> type Client[H commontypes.Head[BLOCK_HASH],

So it takes the commontypes.Head, not the htrktypes.Head.

Reply:
Hmm the reason why HTH is being used here instead of H, is because the Head received from the channel has to be of type HTH to make use of the headtracker operations - `HasChainID`, `IsValid`, `ChainID`. 

https://github.com/smartcontractkit/chainlink/blob/958f677abf6dc045df2ea9eb55e332e44b9458e4/core/chains/evm/headtracker/head_listener.go#L148
https://github.com/smartcontractkit/chainlink/blob/958f677abf6dc045df2ea9eb55e332e44b9458e4/core/chains/evm/headtracker/head_listener.go#L160

Therefore the Client when subscribing to new head has to make use of the same head type as the channel which is HTH.

https://github.com/smartcontractkit/chainlink/blob/958f677abf6dc045df2ea9eb55e332e44b9458e4/core/chains/evm/headtracker/head_listener.go#L224

Therefore the reason why Client is using HTH is because the output (Head) of the channel from Client Method requires method from HTH. Would that be enough to validate the use of HTH for Client?